src/TorchSharp/Tensor/Tensor.Factories.cs

```diff
@@ -2312,23 +2192,56 @@ public static Tensor tensor(IList<(float Real, float Imaginary)> rawArray, long
         /// </summary>
         public static Tensor tensor((float Real, float Imaginary)[,] rawArray, torch.ScalarType? dtype = null, torch.Device? device = null, bool requiresGrad = false)
         {
-            return tensor(rawArray.Cast<(float Real, float Imaginary)>().ToArray(), new long[] { rawArray.GetLongLength(0), rawArray.GetLongLength(1) }, dtype, device, requiresGrad);
+            var dataArray = new float[rawArray.GetLongLength(0), rawArray.GetLongLength(1) * 2];
+            for (var i = 0; i < rawArray.GetLongLength(0); i++) {
+                for (var j = 0; j < rawArray.GetLongLength(0); j++) {
```

Comment:
>0 [](http://example.com/codeflow?start=59&length=1)

should be 1?

Reply:
Well, that's just embarrassing! Thank you for catching it.